compiler/rustc_hir_typeck/src/method/probe.rs

```diff
@@ -1027,6 +1027,15 @@ impl<'a, 'tcx> ProbeContext<'a, 'tcx> {
                     true
                 }
             })
+            // ensure that we don't suggest unstable methods
+            .filter(|candidate| {
+                // note that `DUMMY_SP` is ok here because it is only used for
+                // suggestions and macro stuff which isn't applicable here.
+                !matches!(
+                    self.tcx.eval_stability(candidate.item.def_id, None, DUMMY_SP, None),
```

Comment:
I think this call might be introducing a cycle, which manifests as an `.unwrap()` ICE. Can you check if calling `tcx.lookup_stability(def_id)` also hits it?